src/diskreader.cxx

```diff
@@ -93,6 +93,23 @@ int DiskReader::loadPreferences()
 			LUPPP_NOTE("No default controllers active.");
 		}
 
+		cJSON* hotkeyAssignment = cJSON_GetObjectItem(preferencesJson, "hotkeyAssignment");
+        if (hotkeyAssignment) {
+            keyToGrid.clear();
+            cJSON* key = hotkeyAssignment->child;
+            while (key) {
+                int keycode = std::stoi(key->string);
+                cJSON* gridArray = key;
+                if (cJSON_GetArraySize(gridArray) == 2) {
+                    int x = cJSON_GetArrayItem(gridArray, 0)->valueint;
+                    int y = cJSON_GetArrayItem(gridArray, 1)->valueint;
+                    keyToGrid[keycode] = {x, y};
+                }
+                key = key->next;
+            }
+        } else {
+            LUPPP_WARN("No hotkeyAssignment found in preferences");
+        }
```

Comment:
The indentation here looks off? Perhaps there's a tabs/spaces mixup?

Reply:
Hopefully this is sorted now. This wasnt clear in vs code but is in clion. I fixed some more areas where i messed this up.